java-frontend/src/test/files/se/ConditionAlwaysTrueOrFalseCheckReporting.java

```diff
@@ -0,0 +1,31 @@
+package javax.annotation;
+
+@interface CheckForNull {}
+@interface Nullable {}
+
+public class Class {
+
+  void nestedCondition(boolean a) {
+    if(a) {
+      if(a) { // Noncompliant [[secondary=9]]
+      }
+    }
+  }
+
+  void relationship(boolean a, boolean b) {
+    if(a < b) {
+      if(b > a) { // Noncompliant should have secondary=16
```

Comment:
I don't think we support this syntax, but I can enhance in SONARJAVA-1957 if you wish :)